jax/scipy/special.py

```diff
@@ -150,6 +152,68 @@ def multigammaln(a, d):
   return res + constant
 
 
+# coefs of (2k)! / B_{2k} where B are bernoulli numbers
+# those numbers are obtained using https://www.wolframalpha.com
+_BERNOULLI_COEFS = [
+    12,
+    -720,
+    30240,
+    -1209600,
+    47900160,
+    -1307674368000 / 691,
+    74724249600,
+    -10670622842880000 / 3617,
+    5109094217170944000 / 43867,
+    -802857662698291200000 / 174611,
+    14101100039391805440000 / 77683,
+    -1693824136731743669452800000 / 236364091,
+    186134520519971831808000000 / 657931,
+    -37893265687455865519472640000000 / 3392780147,
+    759790291646040068357842010112000000 / 1723168255201,
+    -134196726836183700385281186201600000000 / 7709321041217,
+]
+
+
+@_wraps(osp_special.zeta)
+def zeta(x, q=None):
+  assert q is not None, "Riemann zeta function is not implemented yet."
+  # Reference: Johansson, Fredrik.
+  # "Rigorous high-precision computation of the Hurwitz zeta function and its derivatives."
+  # Numerical Algorithms 69.2 (2015): 253-270.
+  # https://arxiv.org/abs/1309.2877 - formula (5)
+  # here we keep the same notation as in reference
+  s, a = _promote_args_inexact("zeta", x, q)
+  dtype = lax.dtype(a).type
+  s_, a_ = jnp.expand_dims(s, -1), jnp.expand_dims(a, -1)
+  # precision ~ N, M
+  N = M = dtype(8) if lax.dtype(a) == jnp.float32 else dtype(16)
+  assert M <= len(_BERNOULLI_COEFS)
+  k = np.arange(N, dtype=N.dtype)
+  S = jnp.sum((a_ + k) ** -s_, -1)
+  I = lax.div((a + N) ** (dtype(1) - s), s - dtype(1))
+  T0 = (a + N) ** -s
+  s_over_a = (s_ + np.arange(2 * M, dtype=M.dtype)) / (a_ + N)
+  T1 = jnp.cumprod(s_over_a, -1)[..., ::2]
+  coefs = np.array(_BERNOULLI_COEFS[:T1.shape[-1]], dtype=dtype)
+  T1 = T1 / coefs
+  T = T0 * (dtype(0.5) + T1.sum(-1))
+  return S + I + T
+
+
+@api.custom_jvp
+@_wraps(osp_special.polygamma)
+def polygamma(n, x):
+  assert jnp.issubdtype(lax.dtype(n), jnp.integer)
+  n, x = _promote_args_inexact("polygamma", n, x)
+  dtype = lax.dtype(n).type
+  n_plus = n + dtype(1)
+  sign = dtype(1) - (n_plus % dtype(2)) * dtype(2)
+  return jnp.where(n == 0, digamma(x), sign * jnp.exp(gammaln(n_plus)) * zeta(n_plus, x))
+# FIXME: this does not work for higher derivatives, should we create
+# a new primitive for polygamma?
+polygamma.defjvps(None, lambda g, ans, n, x: lax.mul(g, polygamma(n + 1, x)))
+
+
```

Comment:
Please add `polygamma` and `zeta` to the documentation, by listing them in https://github.com/google/jax/blob/master/docs/jax.scipy.rst